prov/efa/src/rdm/rxr_ep.c

```diff
@@ -2283,6 +2355,17 @@ int rxr_endpoint(struct fid_domain *domain, struct fi_info *info,
 	rxr_ep->efa_max_outstanding_rx_ops = efa_domain->device->rdm_info->rx_attr->size;
 	rxr_ep->efa_device_iov_limit = efa_domain->device->rdm_info->tx_attr->iov_limit;
 
+	api_version = efa_domain->util_domain.fabric->fabric_fid.api_version;
+	if (FI_VERSION_LT(api_version, FI_VERSION(1,18))) {
+		/* for API<1.18 we get use_device_rdma from the environment as
+		   it was queried during domain open. */
+		rxr_ep->use_device_rdma = efa_domain->use_device_rdma_api_before_118;
```

Comment:
It seems that we need not only the value of `use_device_rdma` and whether user set it explicitly. Can we check whether user set the envar explicitly on the fly? Then we do not need to save that in `rxr_ep`.

Reply:
small behavioral change for API 1.17 after this update: the environment variable would be read later and could change between enabling several endpoints.  I assume this is not a concern, and so we could remove the fetch during domain open.

Reply:
Note that behavior of fetch envar during domain open was introduced very recently, and was not part of any release.